doc/docs.md

```diff
@@ -1538,6 +1538,27 @@ println(s)
 // "odd"
 ```
 
+Similar to Rust's `if let`, a value can be bound to a variable within an if statement.
+
+```v
+m := {
+	'foo': 'bar'
+}
+
+if v := m['foo'] {
+	println(v) // bar
+} else {
+	println('not found')
+}
+
+res := if v := m['foobar'] {
+	v
+} else {
+	'baz'
+}
```

Comment:
imho, that is not a good illustration and may be more confusing than helping, since you can do the same in a one liner: `res := m['foobar'] or { 'baz' }` .

Reply:
Agreed.

Reply:
True. Some further operations on that variable to would have made it more real world.
```v
res := if v := m['foobar'] {
	v.title()
} else {
	'Baz'
}
```

Your example below is good. Thanks!

Reply:
What about something like this?
```v
struct User {
	name string
}

arr := [User{'John'}]

u_name := if v := arr[0] {
	v.name
} else {
	'Unnamed'
}
println(u_name) // John
```

Reply:
Yes, that example is more useful.